boot/zephyr/arm_cleanup.c

```diff
@@ -33,4 +38,18 @@ __weak void z_arm_clear_arm_mpu_config(void)
 		ARM_MPU_ClrRegion(i);
 	}
 }
+#elif CONFIG_CPU_HAS_NXP_MPU
+__weak void z_arm_clear_arm_mpu_config(void)
```

Comment:
Wouldn't it make more sense to have this function in the Zephyr tree? Is `CONFIG_CPU_HAS_ARM_MPU` also defined to NXP?

Reply:
Yes, that could make sense - but the same could be set for the existing `z_arm_clear_arm_mpu_config()` in the same file? `CONFIG_CPU_HAS_ARM_MPU` is not defined for the NXP MPUs.

Reply:
> Yes, that could make sense - but the same could be said for the existing `z_arm_clear_arm_mpu_config()` in the same file?

Which is itself a copy of https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/arm/core/aarch32/cortex_m/scb.c#L47, I assume.

Reply:
z_arm_clear_arm_mpu_config() is implemented here as I encounter  the case wen mcuboot have to chainnload a legacy none-zephyr application.

Reply:
So normally chain loaded application should does MPU cleanup. @henrikbrixandersen Shouldn't this be moved to the app? Or you encountered similar case as I had?

Reply:
No. The issue, I was facing, was a Zephyr-based MCUboot built with `CONFIG_HW_STACK_PROTECTION=y` where the application would sometimes (depending on the stack sizes used in the application) crash with an MPU fault even before reaching the NXP MPU setup code of the application image.

Disabling the MPU and clearing the MPU regions prior to booting the application solves this issue.

Reply:
Name which match zephyr's analog function for NXP mpu be used here instead of `z_arm_clear_arm_mpu_config`. Otherwise this function will be overridden by implementation dedicated to ARM MPU.

Reply:
BTW - there is no a NXP MPU clear function in the zephyr-rtos right now.

Reply:
I would change function name name for reflecting that it works on NXP mpu.

Reply:
> No. The issue, I was facing, was a Zephyr-based MCUboot built with `CONFIG_HW_STACK_PROTECTION=y` where the chain-loaded application would sometimes (depending on the stack sizes used in the application) crash with an MPU fault even before reaching the NXP MPU setup code of the application image.
> 
> Disabling the MPU and clearing the MPU regions prior to booting the application solves this issue.

@henrikbrixandersen the chain-loaded application could enable CONFIG_INIT_ARCH_HW_AT_BOOT, that would do the clearing during early Zephyr boot. Or are you assuming an non-Zephyr-based chain-loaded firmware?